src/hooks/private/useRadioGroup/useRadioGroup.ts

```diff
@@ -39,44 +39,44 @@ export function useRadioGroup<ValueType extends string = string>(
     } = props;
 
     const controlId = useUniqId();
-    const [valueState, setValueState] = React.useState(
-        defaultValue ?? options[0]?.value?.toString(),
+    const [currentValue, setValueState] = useControlledState<string | null, ValueType>(
+        value,
+        // FIXME: Do not set defaultValue to first option value
+        defaultValue ?? options[0]?.value ?? null,
+        onUpdate,
     );
-    const isControlled = typeof value !== 'undefined';
-    const currentValue = isControlled ? value : valueState;
 
-    const handleChange = React.useCallback(
-        (event: React.ChangeEvent<HTMLInputElement>) => {
-            if (!isControlled) {
-                setValueState(event.target.value);
-            }
-            if (onChange) {
-                onChange(event);
-            }
-            if (onUpdate) {
-                onUpdate(event.target.value as ValueType);
-            }
-        },
-        [isControlled, onUpdate, onChange],
-    );
+    const fieldRef = useFormResetHandler({
+        initialValue: currentValue as ValueType,
+        onReset: setValueState,
+    });
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setValueState(event.target.value as ValueType);
+
+        if (onChange) {
+            onChange(event);
+        }
+    };
 
     const containerProps = {
+        ...filterDOMProps(props, {labelable: true}),
         role: 'radiogroup',
         'aria-disabled': disabled,
-        'aria-label': props['aria-label'],
-        'aria-labelledby': props['aria-labelledby'],
     };
 
-    const optionsProps = options.map((option) => ({
+    const optionsProps: OptionsProps<ValueType>[] = options.map((option) => ({
         name: name || controlId,
         value: option.value,
         content: option.content,
         title: option.title,
         checked: currentValue === String(option.value),
         disabled: disabled || option.disabled,
         onChange: handleChange,
+        // FIXME: onFocus and onBlur should be on the container
```

Comment:
Will it change anything?

Reply:
Yes, now onFocus/onBlur is triggered when switching between radio inputs within a group.

Reply:
No if use `useFocusWithin` hook